apps/anoma_node/lib/node/transaction/shard_supervisor.ex

```diff
@@ -0,0 +1,256 @@
+defmodule Anoma.Node.Transaction.ShardSupervisor do
+  @moduledoc """
+  I am the supervisor for `Anoma.Node.Transaction.Shard` processes.
+
+  I start and manage individual `Shard` processes according to a strategy and
+  schema provided in my arguments. I create and maintain a named ETS table
+  (`:shard_key_map`) mapping keys to the registered name
+  (`{:via, Registry, {Anoma.Node, {Shard, key}}}`) of the `Shard` process
+  responsible for that key. The actual lookup of keys is handled by the
+  `Anoma.Node.Transaction.Ordering`.
+
+  ### Key Concepts
+
+  - **Supervisor Args:** Keyword list including `:strategy` and `:schema`.
+  - **Strategy:** Determines how shards are created (e.g., `:one_per_key`).
+  - **Schema:** Defines the initial keys and their starting values.
+  - **ETS Table:** `:shard_key_map` for key -> shard name lookup (used by Anoma.Node.Transaction.Ordering).
```

Comment:
Why are we using ETS tables, and not using our `Anoma.Node.Tables` module? We need support for rocksdb, as ETS tables have massive flaw and we should be operate where we can turn them automatically into rocksdb tables with mnesia as the interface